jsk_panda_robot/panda_eus/euslisp/dual_panda-interface.l

```diff
@@ -0,0 +1,380 @@
+      req)
+     ))
+  (:set-all-joint-pd-gain
+   (pgain dgain)
+   (dolist (j (send robot :joint-list))
+     (send self :set-joint-pd-gain (send j :name) pgain dgain))
+   )
+  (:check-error ()
+                (ros::spin-once "error_group")
+                (or r-error l-error)
+                )
+  (:callback-rarm-error (msg)
+                        (setq r-error (send msg :data))
+                        )
+  (:callback-larm-error (msg)
+                        (setq l-error (send msg :data))
+                        )
+  (:wait-recover-error () (send error-recovery-act :wait-for-result))
+  (:recover-error (&key (wait t))
+                  (let ((goal (instance franka_msgs::ErrorRecoveryActionGoal :init)))
+                    (send goal :header :stamp (ros::time-now))
+                    (send error-recovery-act :send-goal goal)
+                    (if wait (send self :wait-recover-error))
+                    ))
+  ;; gripper action for real-controller
+  (:send-gripper-grasp-action
+   (act width speed force &key (wait t) (inner 0.005) (outer 0.07))
+   (let ((goal (instance franka_gripper::GraspActionGoal :init)))
+     (send goal :header :stamp (ros::time-now))
+     (send goal :goal :width width) ;; [m]
+     (send goal :goal :speed speed) ;; [m/s]
+     (send goal :goal :force force) ;; [N]
+     (send goal :goal :epsilon :inner inner) ;; [m]
+     (send goal :goal :epsilon :outer outer) ;; [m]
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:send-gripper-homing-action
+   (act &key (wait t))
+   (let ((goal (instance franka_gripper::HomingActionGoal :init)))
+     (send goal :header :stamp (ros::time-now))
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:send-gripper-move-action
+   (act width speed &key (wait t))
+   (let ((goal (instance franka_gripper::MoveActionGoal :init)))
+     (send goal :header :stamp (ros::time-now))
+     (send goal :goal :width width) ;; [m]
+     (send goal :goal :speed speed) ;; [m/s]
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:send-gripper-stop-action
+   (act &key (wait t))
+   (let ((goal (instance franka_gripper::StopActionGoal :init)))
+     (send goal :header :stamp (ros::time-now))
+     ;;
+     (send act :send-goal goal)
+     (if wait (send act :wait-for-result))
+     ))
+  (:stop-gripper
+   (arm &key (wait nil))
+   (let (acts)
+     (case
+      arm
+      (:rarm (setq acts (list r-gripper-stop-action)))
+      (:larm (setq acts (list l-gripper-stop-action)))
+      (:arms (setq acts (list r-gripper-stop-action l-gripper-stop-action)))
+      )
+     (when acts
+       (dolist (act acts)
+         (send self :send-gripper-stop-action act
+               pos (/ (* 1000 0.08) tm) :wait nil))
+       (if wait (mapcar #'(lambda (act) (send act :wait-for-result)) acts))
+       )))
+  (:homing-gripper
```

Comment:
are you intend this method as Public? if so add function documentation, simple one is  ok. In that sense, if we had function documentation on "start-grasp, stop-grasp, angle-vector,,, ", that would help users to understand which method should we use.

Reply:
Sorry, I'll add documentation.

Just FYI: `homing` means `initialization` in panda gripper. This is a panda-specific method.

Reply:
Documentation was added